### PR TITLE
GT-1229: Update PageView Analytics to Have Proper ScreenName and SiteSection Values

### PR DESCRIPTION
**Pull request by reldredge71**
Ann wanted to make sure that every screen change analytics event contained a value for the "site section" property.  I made those changes, but then took it a step further and refactored the way analyticsScreenName, analyticsSiteSection, and analyticsSiteSubSection are defined throughout the app.  Let me know what you think.

Also, here is a spreadsheet laying out all of the expected values for the analytics events -- Look specifically at the tab titled "iOS TrackState Notes": https://docs.google.com/spreadsheets/d/1YLkGQ8ptXWn6xmB82kzUYIkXHsI-BsnA2iQanQdPcnw/edit#gid=2099914398

### CODE REVIEW

**Comment by frett** [comment]
@reldredge71 a bunch of merge conflicts, probably due to Levi merging in the multiplatform wrapper cleanup PR

**Comment by levieggertcru** [comment]
Thank you @reldredge71 for going through and double checking those values and cleaning that up some more.

File: godtools/App/Features/ShareTool/ShareTool/ShareToolViewModel.swift [hunk 1]
```diff
@@ -36,14 +36,24 @@ class ShareToolViewModel: ShareToolViewModelType {
...
         
-        let analyticsScreenName: String = resource.abbreviation + "-" + String(pageNumber)
-        
-        analytics.pageViewedAnalytics.trackPageView(trackScreen: TrackScreenModel(screenName: analyticsScreenName, siteSection: resource.abbreviation, siteSubSection: ""))
+        analytics.pageViewedAnalytics.trackPageView(trackScreen: TrackScreenModel(screenName: analyticsScreenName, siteSection: analyticsSiteSection, siteSubSection: analyticsSiteSubSection))
                 
-        analytics.trackActionAnalytics.trackAction(trackAction: TrackActionModel(screenName: analyticsScreenName, actionName: AnalyticsConstants.Values.shareIconEngaged, siteSection: resource.abbreviation, siteSubSection: "", url: nil, data: [
-            AnalyticsConstants.ActionNames.shareIconEngagedCountKey: 1
+        analytics.trackActionAnalytics.trackAction(trackAction: TrackActionModel(screenName: analyticsScreenName, actionName: AnalyticsConstants.ActionNames.shareIconEngaged, siteSection: analyticsSiteSection, siteSubSection: analyticsSiteSubSection, url: nil, data: [
+            AnalyticsConstants.Keys.shareAction: 1
```
Comment (levieggertcru) [inline godtools/App/Features/ShareTool/ShareTool/ShareToolViewModel.swift:56]: Hey @reldredge71 looks like you fixed the value name for this analytics?

Comment (reldredge71) [reply godtools/App/Features/ShareTool/ShareTool/ShareToolViewModel.swift:56]: @levieggertcru I noticed that `Keys.shareAction` already had the same value as `ActionNames.shareIconEngagedCountKey`, so I just decided to keep one and remove the other.  Plus, I wanted to clean up `AnalyticsConstants` and make it easier to understand; I wanted `ActionNames` to be the names of the `trackAction` events, and I wanted `Keys` to be the keys of the `data` object being passed to the analytics events (basically, everything that Cru tracks and has the "cru." prefix).   I didn't know that `ActionNames.shareIconEngagedCountKey` needed fixing, what should it be?

Comment (levieggertcru) [reply godtools/App/Features/ShareTool/ShareTool/ShareToolViewModel.swift:56]: Hey @reldredge71 nothing needs to be fixed, I thought the key was incorrect and the change here was addressing that, but now I see it was duplicated as you said.  This looks good to me.

**Review by levieggertcru — APPROVED** [review]

**Pull request merged**